bin/lang-js/BUCK

```diff
@@ -1,113 +1,84 @@
-    "prettier_check",
-    "typescript_check",
-    "typescript_dist",
-    "typescript_runnable_dist",
-    "typescript_runnable_dist_bin",
 )
 load(
-    "@prelude-si//:pnpm.bzl",
-    "pnpm_task_library",
-    "pnpm_task_binary",
-    "pnpm_task_test",
+    "@prelude-si//:deno.bzl",
+    "deno_compile",
+    "deno_format",
+    "deno_test",
 )
 
 export_file(
     name = "package.json",
 )
 
-package_node_modules(
-    name = "node_modules",
-)
-
-filegroup(
-    name = "src",
-    srcs = glob([
-        "src/**/*.ts",
-        "tsconfig.json",
-    ]),
-)
-
-filegroup(
-    name = "test_src",
-    srcs = glob([
-        "tests/**/*",
-    ]),
-)
-
-prod_deps_srcs = {
-    "lib/ts-lib": "//lib/ts-lib:src",
-}
-
-dev_deps_srcs = {
-    "lib/eslint-config": "//lib/eslint-config:src",
-    "lib/tsconfig": "//lib/tsconfig:src",
-}
-
-npm_bin(
-    name = "tsup",
-)
-
-typescript_dist(
-    name = "dist",
-    srcs = [":src"],
-    tsup = ":tsup",
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+export_file(
+    name = "deno.json",
 )
 
-typescript_runnable_dist(
+alias(
     name = "lang-js",
+    actual = ":bin"
 )
 
-typescript_runnable_dist_bin(
+deno_compile(
     name = "bin",
-    typescript_runnable_dist = ":lang-js",
+    main = "src/index.ts",
+    out = "lang-js",
+    srcs = glob([
+        "src/**/*.ts",
+        "src/**/*.js",
+    ]) +
+    ["//lib/ts-lib-deno:ts-lib-deno"],
+    permissions = [
+        "allow-all",
+    ],
+    unstable_flags = [
+        "worker-options",
+    ],
+    visibility = ["PUBLIC"],
 )
 
-eslint(
-    name = "check-lint",
-    directories = ["src", "tests"],
-    srcs = [":src", ":test_src"] + glob([".eslint*"]),
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+deno_format(
+    name = "fix-format",
+    srcs = glob(["**/*.ts", "**/*.js"]),
+    ignore = ["node_modules"],
 )
 
-typescript_check(
-    name = "check-type",
-    srcs = [":src"],
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+deno_format(
+    name = "check-format",
+    srcs = glob(["**/*.ts", "**/*.js"]),
+    check = True,
 )
 
-ts_test(
+deno_test(
     name = "test-unit",
-    srcs = [":src", ":test_src"],
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+    srcs = glob(["**/tests/*.spec.ts"]),
+    ignore = ["node_modules"],
+    permissions = [
+        "allow-all",
+    ],
+    unstable_flags = [
+        "worker-options",
+    ],
 )
 
+dev_deps_srcs = {
+    "lib/eslint-config": "//lib/eslint-config:src",
+    "lib/tsconfig": "//lib/tsconfig:src",
+}
+
 nix_omnibus_pkg(
     name = "omnibus",
     pkg_name = "lang-js",
     build_dep = "//bin/lang-js:bin",
-)
-
-pnpm_task_binary(
-    name = "docs",
-    command = "docs",
-    srcs = glob(["src/**/*"]),
-    path = "bin/lang-js",
-    deps = [
-        "//:node_modules",
-    ],
-    visibility = ["PUBLIC"],
+    srcs = {
+        "//:deno.json": ".",
```

Comment:
I was curious so see that `//:deno.lock` wasn't in the root `BUCK` file, so I presume that the lock file isn't needed to build?